src/main/kotlin/org/rust/lang/core/types/ty/TyPrimitive.kt

```diff
@@ -15,9 +18,14 @@ import org.rust.lang.core.psi.RsPath
  * tuples or arrays as primitive.
  */
 abstract class TyPrimitive : Ty() {
-
     abstract val name: String
 
+    protected open fun copyWithAlias(aliasedBy: BoundElement<RsTypeAlias>?): Ty = this
+
+    override fun withAlias(aliasedBy: BoundElement<RsTypeAlias>): Ty = copyWithAlias(aliasedBy)
+
+    override fun superFoldWith(folder: TypeFolder): Ty = copyWithAlias(aliasedBy?.foldWith(folder))
```

Comment:
This could be implemented using `withAlias` like `if (aliasedBy == null) this else withAlias(aliasedBy.foldWith(folder))` so that introducing a new method `copyWithAlias` is not needed.